micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -126,6 +126,40 @@ impl NVSStorage {
     }
 }
 
+impl StorageDiagnostic for NVSStorage {
+    type Error = NVSStorageError;
+    fn log_space_diagnostic(&self) -> Result<(), Self::Error> {
+        let part_name = CString::new("nvs")
+            .map_err(|_| EspError::from_non_zero(NonZeroI32::new(ESP_ERR_INVALID_ARG).unwrap()))?;
+        let mut stats: nvs_stats_t = Default::default();
+        esp!(unsafe { nvs_get_stats(part_name.as_ptr(), &mut stats as *mut _) })?;
+
+        // 32 bytes per entry
+        let used_entries = stats.used_entries;
+        let used_space = used_entries * 32;
+        let used_page_header_space = (used_entries % 125) * 32;
```

Comment:
could we make these `const` with descriptive names? can't tell what 125 is for

Reply:
I was trying to account for entries dedicated to NVS page headers but it seems that the total entries allowed does include those in the count, so I'm removing this anyway